lib/hdf5-db/jhdf5-db-shared.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * JULEA - Flexible storage framework
+ * Copyright (C) 2019 Benjamin Warnke
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * \file
+ **/
+#ifndef JULEA_DB_HDF5_SHARED_C
+#define JULEA_DB_HDF5_SHARED_C
+#include <julea-config.h>
+#include <julea.h>
+#include <julea-db.h>
+#include <julea-object.h>
+#include <glib.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <string.h>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#pragma GCC diagnostic ignored "-Wunused-function"
```

Comment:
Warnings have to be fixed, not disabled.

Reply:
The plugin interface defines lots of parameters, which are just unused at the moment. I can either flag each parameter on its own as unused, or just let the compiler throw around 100 warnings of unused parameters.

Reply:
The unused function thing can be removed